File: data/storeinfo.go
package data

type StoreInfo struct {
	Id          string                    `json:"id"`   // Store ID
	Name        string                    `json:"name"` // Store name
	Description string                    `json:"desc"` //Store Description
	IconCID     string                    `json:"iconcid"`
	Packages    map[string]PackagePreview `json:"packages"`
}

File: publisher/frontend/src/data/storeInfo.ts
import {PackageInfo } from "./packageInfo"

export interface StoreInfo {
    id: string;
    desc: string;
    name: string;
    iconcid: string;
    packages?: { [key:string]: PackageInfo };
}


export function updatePackage(info: StoreInfo, pkg: PackageInfo): void {
    //info.packages = {}
    if (!info.packages)
        info.packages = {}
    info.packages[pkg.id] = pkg
    //console.log("(((((((((((((((222222)))))))))))))))",info)
}